### PR TITLE
feat: Add wrapper for ROOT hadd

### PR DESCRIPTION
**Pull request by GoodingJamie**
Adds a wrapper for the ``hadd`` utility in the ROOT software package used primarily in High Energy Physics. Since this is the first physics wrapper, the top-level category ``phys`` is also added here.

### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* the `environment.yaml` pinning has been updated by running `snakedeploy pin-conda-envs environment.yaml` on a linux machine,
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

File: phys/root/hadd/test/Snakefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+rule hadd:
+    input:
+        expand("ntuple{n}.root", n=range(3)),
+    log:
+        "logs/hadd/hadd.log",
+    params:
+        "-j {threads}",
+    output:
+        "ntuple.root",
```
Comment (fgvieira) [inline phys/root/hadd/test/Snakefile:9]:
```suggestion
    threads: 2
    output:
        "ntuple.root",
```

File: phys/root/hadd/wrapper.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+__copyright__ = "Copyright 2024, Jamie Gooding"
+__email__ = "jamie.gooding@cern.ch"
+__license__ = "MIT"
+
+
+from snakemake.shell import shell
+
+log = snakemake.log_fmt_shell()
+
+shell("(hadd {snakemake.params} {snakemake.output} {snakemake.input}) {log}")
```
Comment (fgvieira) [inline phys/root/hadd/wrapper.py:11]:
```suggestion
shell("hadd -j {snakemake.threads} {extra} {snakemake.output} {snakemake.input} {log}")
```

File: phys/root/hadd/wrapper.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+__author__ = "Jamie Gooding"
+__copyright__ = "Copyright 2024, Jamie Gooding"
+__email__ = "jamie.gooding@cern.ch"
+__license__ = "MIT"
+
+
+from snakemake.shell import shell
+
+log = snakemake.log_fmt_shell()
```
Comment (fgvieira) [inline phys/root/hadd/wrapper.py:9]:
```suggestion
extra = snakemake.params.get("extra", "")
log = snakemake.log_fmt_shell()
```

File: phys/root/hadd/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+url: https://root.cern/doc/master/hadd_8cxx.html
+authors:
+  - Jamie Gooding
+input:
+  - ROOT files to be combined.
+output:
+  - >
+    single ROOT file containing contents of input files.
+params:
+  - parameters of hadd defined in the ROOT documentation, e.g. -j N for multiple cores, -f[0-9] for compression levels
```
Comment (fgvieira) [inline phys/root/hadd/meta.yaml:12]:
```suggestion
  - extra: extra parameters to `hadd` (see ROOT documentation, e.g. `-f[0-9]` for compression levels)
```

File: phys/root/hadd/test/Snakefile [hunk 5]
```diff
@@ -0,0 +1,12 @@
+rule hadd:
+    input:
+        expand("ntuple{n}.root", n=range(3)),
+    log:
+        "logs/hadd/hadd.log",
+    params:
+        extra="",
```
Comment (fgvieira) [inline phys/root/hadd/test/Snakefile:7]:
```suggestion
        extra="-f5",
```

**Comment by GoodingJamie** [comment]
Thanks a lot for the review @fgvieira 😄

**Comment by fgvieira** [comment]
Sure! Can you just add the tests to `test.py`?
Otherwise it looks quite good.

**Comment by GoodingJamie** [comment]
I've added the test to ``test.py`` (which passes in CI) and included the pinned environment. From my side this is good to go 😄

File: phys/root/hadd/environment.linux-64.pin.txt [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+        ╚═╝     ╚═╝╚═╝  ╚═╝╚═╝     ╚═╝╚═════╝ ╚═╝  ╚═╝
+
+        mamba (0.13.0) supported by @QuantStack
+
+        GitHub:  https://github.com/mamba-org/mamba
+        Twitter: https://twitter.com/QuantStack
+
+█████████████████████████████████████████████████████████████
+
+# This file may be used to create an environment using:
```
Comment (fgvieira) [inline phys/root/hadd/environment.linux-64.pin.txt:24]:
```suggestion
# This file may be used to create an environment using:
```

**Review by fgvieira — APPROVED** [review]

**Comment by fgvieira** [comment]
Cool, and thanks for your contribution!

**Pull request merged**